src/NonlinearSolve.jl

```diff
@@ -31,22 +31,23 @@ end
 include("utils.jl")
 include("jacobian.jl")
 include("raphson.jl")
+include("trustRegion.jl")
 include("ad.jl")
 
 import SnoopPrecompile
 
 SnoopPrecompile.@precompile_all_calls begin for T in (Float32, Float64)
     prob = NonlinearProblem{false}((u, p) -> u .* u .- p, T(0.1), T(2))
-    for alg in (NewtonRaphson,)
-        solve(prob, alg(), abstol = T(1e-2))
+    for alg in (NewtonRaphson(), TrustRegion(T(10.0)))
```

Comment:
How is the max trust radius generally automated? Most other packages don't require this.

Reply:
When I've read about this it often seems quite ad-hoc. E.g. use fixed values for the specific problem or something like $||f(x)||=\Delta_{\text{max}}$ or $max(x) - min(x)$.

Reply:
But I'll look into it!

Reply:
I think that's fine. Whatever value people would put in here would be more ad-hoc. I just wrote 10.0 because it seemed like a good number 😅